PyInquirer/prompts/list.py

```diff
@@ -62,6 +62,9 @@ def _init_choices(self, choices, default=None):
                 if searching_first_choice:
                     self.selected_option_index = i  # found the first choice
                     searching_first_choice = False
+                if default and (default == i or default == c):
+                    self.selected_option_index = i  # default choice exists
+                    searching_first_choice = False
```

Comment:
this will only support a non-integer type of `default` when `c` is of type `basestring`. if `c` is of the form `{"name": "_", "value": "_"}` the `default == c` will not work.

How about something like this:

```python
self.choices = []  # list (name, value, disabled)
# selection is 'default' if 'default' is of type int
self.selected_option_index = default if isinstance(default, int) else 0
# if default is not a int it needs to be found in values field of the choices
search_for_value = isinstance(default, basestring)
# used to indicate if the non-int value of default was in fact found
found_default_value = False
for i, c in enumerate(choices):
    if isinstance(c, Separator):
        name, value, disabled = c, None, None
    elif isinstance(c, basestring):
        name, value, disabled = c, c, None
    elif isinstance(c, dict):
        name = c.get('name')
        value = c.get('value', name)
        disabled = c.get('disabled', None)
    else:
        raise ValueError("Unsupported 'choice' type: " + type(c))

    self.choices.append((name, value, disabled))

    if search_for_value and default == value:
        self.selected_option_index = i
        found_default_value = True

if search_for_value and not found_default_value:
    choices = [v for _, v, _ in self.choices]
    raise ValueError(f"default: '{0}' not found in choices: {1}".format(default, choices))

```

Reply:
Lemme take a look at this

Reply:
Any update on this?

Reply:
sorry to bug, but this would be super useful for my work.